assets/js/invidious_iframe_api.js

```diff
@@ -0,0 +1,386 @@
+                    for(var x=0;x<promise_array.length;x++){
+                        if(message.data.command==='getvolume'){
+                            promise_array[x](message.data.value*100);
+                        }
+                        else{
+                            promise_array[x](message.data.value);
+                        }
+                    }
+                    this.message_wait[message.data.command] = [];
+                    break;
+                case 'event':
+                    if(message.data.eventname!==undefined&&typeof message.data.eventname==='string'){
+                        this.event_executor(message.data.eventname);
+                        var previous_status = this.player_status;
+                        switch(message.data.eventname){
+                            case 'ended':
+                                this.player_status = 0;
+                                break;
+                            case 'play':
+                                this.player_status = 1;
+                                break;
+                            case 'timeupdate':
+                                this.player_status = 1;
+                                this.eventdata = Object.assign({},this.eventdata,message.data.value);
+                                break;
+                            case 'pause':
+                                this.player_status = 2;
+                                break;
+                            case 'waiting':
+                                this.player_status = 3;
+                                break;
+                            case 'loadedmetadata':
+                                this.eventdata = Object.assign({},this.eventdata,message.data.value);
+                                break;
+                        }
+                        if(previous_status!==this.player_status){
+                            this.event_executor('statechange');
+                        }
+                    }
+            }
+        }
+    }
+    promise_resolve(){
+        var res_outer;
+        var pro = new Promise((res,rej)=>{res_outer=res});
+        return {'promise':pro,'resolve':res_outer};
+    }
+    promise_send_event(event_name){
+        if(invidious_embed.api_promise){
+            var pro_object = this.promise_resolve();
+            this.message_wait[event_name].push(pro_object.resolve);
+            this.postMessage({eventname:event_name});
+            return pro_object.promise;
+        }
+        else{
+            return this.eventdata[event_name];
+        }
+    }
+    getPlayerState(){
+        return this.player_status;
+    }
+    playVideo(){
+        this.postMessage({eventname:'play'});
+    }
+    pauseVideo(){
+        this.postMessage({eventname:'pause'});
+    }
+    getVolume(){
+        return this.promise_send_event('getvolume');
+    }
+    setVolume(volume){
+        volume = Number(volume);
```

Comment:
According API reference, volume had to be Number. And if YT is not accept strings, we also should not. Or at least, `console.warn` something about this.